tc/core/polyhedral/mapping_types.h

```diff
@@ -28,6 +28,17 @@ struct MappingId : public isl::id {
  public:
   MappingId(const MappingId& id) : isl::id(id), dim(id.dim) {}
 
+  // Is "id" a mapping of the type provided as template argument?
+  template <typename MappingType>
+  bool isOfType() const {
```

Comment:
I would go with just `is`, which is sort of consistent with `as` that we use for conversions. 

We could also have an `as` here and return a null id if the type is wrong (given that we have implicit conversion to bool), but I don't like relying on implicit conversions.